### PR TITLE
Add `--backup-working-dir` option

### PR DESCRIPTION
**Pull request by ffuugoo**
Add an option to backup working dir before each restart. One of the data consistency bugs we fixed in Qdrant earlier, actually happened during data recovery on startup, so this option should help catch similar bugs.

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+use tokio::fs;
+
+pub fn copy_dir(
+    src: impl Into<PathBuf>,
+    dst: impl Into<PathBuf>,
+) -> BoxFuture<'static, anyhow::Result<()>> {
+    copy_dir_impl(src.into(), dst.into()).boxed()
+}
+
+async fn copy_dir_impl(src: PathBuf, dst: PathBuf) -> anyhow::Result<()> {
```
Comment (ffuugoo) [inline src/util.rs:13]: There's like million lines of `with_context` in this function, but debugging recursive copy errors is the most annoying thing ever, so I stand by this.

**Comment by ffuugoo** [comment]
I'm done for the day, but I'll add an option to enable `--backup-working-dir` to `crash-things.sh` tomorrow morning.

I do not yet plan to enable it on CI, though. There's no point in doing so if we don't save/snapshot storage state on CI failures. (And there's probably no point enabling it on CI *by default* in general.)

I just want to have this feature ready, before I'll try to repro the missing payload issue once again, so that I can collect all possible info on the first run.

**Comment by ffuugoo** [comment]
@agourlay I think I'm node with this. Ready for review.

**Review by tellet-q — APPROVED** [review]
lgtm

**Review by agourlay — APPROVED** [review]

**Pull request merged**